tensorflow_estimator/python/estimator/estimator.py

```diff
@@ -855,9 +855,10 @@ def _export_all_saved_models(
         raise ValueError("Couldn't find trained model at %s." % self._model_dir)
 
       export_dir = export_helpers.get_timestamped_export_dir(export_dir_base)
-      temp_export_dir = export_helpers.get_temp_export_dir(export_dir)
+      if gfile.NeedsTempLocation(export_dir):
+        export_dir = export_helpers.get_temp_export_dir(export_dir)
```

Comment:
Rather than modify this in place, let's rename the vars final_export_dir and curr_export_dir or similar, to make it clearer what is happening, and to remove the need to re-compute the final export dir before the rename below.